src/jsonpath/json_path.rs

```diff
@@ -45,22 +45,23 @@ impl<'i> Query<'i> {
     /// ($.foo)
     #[allow(dead_code)]
     pub fn pop_last(&mut self) -> Option<(String, JsonPathToken)> {
-        let last = self.root.next_back();
-        match last {
-            Some(last) => match last.as_rule() {
-                Rule::literal => Some((last.as_str().to_string(), JsonPathToken::String)),
-                Rule::number => Some((last.as_str().to_string(), JsonPathToken::Number)),
-                Rule::numbers_list => {
-                    let first_on_list = last.into_inner().next();
-                    first_on_list.map(|first| (first.as_str().to_string(), JsonPathToken::Number))
-                }
-                Rule::string_list => {
-                    let first_on_list = last.into_inner().next();
-                    first_on_list.map(|first| (first.as_str().to_string(), JsonPathToken::String))
-                }
-                _ => panic!("pop last was used in a none static path"),
-            },
-            None => None,
+        let last = self
+            .root
+            .next_back()
+            .expect("pop last was used in a none static path");
```

Comment:
When `None` it was returning `None` and not panicking?
(before there was 1 `panic` and now there are `expect` and `panic`)

Reply:
@oshadmi good catch not sure why I did it